src/Symfony/Component/HttpKernel/EventListener/AbstractSessionListener.php

```diff
@@ -160,7 +161,7 @@ public function onKernelResponse(ResponseEvent $event)
                     $sessionCookieHttpOnly,
                     $sessionCookieSameSite
                 );
-            } elseif ($sessionId !== $requestSessionCookieId) {
+            } elseif ($sessionId !== $requestSessionCookieId && !$isSessionEmpty) {
```

Comment:
maybe we should even not *write* the session in the backend in this case?

Reply:
If a value is removed we need to write it to remove that value from the session bag.